File: medium/2390. Removing Stars From a String/2390. Removing Stars From a String.py
class Solution:
    def removeStars(self, s: str) -> str:
        # res = ""

        # for i in range(len(s)):
        #     if s[i] == "*":
        #         res = res[:-1]
        #     else:
        #         res += s[i]

        # return res

        res = []
        for v in s:
            if v != "*":
                res.append(v)
            else:
                res.pop()
        return "".join(res)


solution = Solution()
print(solution.removeStars(s="leet**cod*e"))
print(solution.removeStars(s="erase*****"))

File: medium/2390. Removing Stars From a String/2390. Removing Stars From a String.go
package main

func removeStars(s string) string {
	res := []byte{}

	for i := 0; i < len(s); i++ {
		if s[i] != '*' {
			res = append(res, s[i])
		} else {
			res = res[:len(res)-1]
		}
	}

	return string(res)
}
